internal/controller/controlplane/helper.go

```diff
@@ -43,14 +43,15 @@ import (
 
 const (
 	etcdMemberConditionTypeJoined = "Joined"
+	generatedMachineRoleLabel     = "cluster.x-k8s.io/generateMachine-role"
 )
 
 func (c *K0sController) createMachine(ctx context.Context, name string, cluster *clusterv1.Cluster, kcp *cpv1beta1.K0sControlPlane, infraRef corev1.ObjectReference, failureDomain *string) (*clusterv1.Machine, error) {
 	machine, err := c.generateMachine(ctx, name, cluster, kcp, infraRef, failureDomain)
 	if err != nil {
 		return nil, fmt.Errorf("error generating machine: %w", err)
 	}
-	_ = ctrl.SetControllerReference(kcp, machine, c.Scheme)
+	_ = ctrl.SetControllerReference(kcp, machine, c.Client.Scheme())
```

Comment:
Out of curiosity, why do we change it?

Reply:
writing tests for the controller I saw that we access the schema at this point and I thought about refactoring it using the client schema directly and remove `K0scontroller.Scheme` field. Both are obtained from `manager.Manager` so I thought it was safe to make this change